### ISSUE TITLE
select2: search for items empty when allowNewItems is set to false

### ISSUE THREAD

**Issue by pbauer** [plone/mockup#516]
When I use select2 for a field and disable allowNewItems the search is broken.

Steps to reproduce:

```python
accountable = schema.Choice(
    title=_(u'Inhaber'),
    vocabulary='plone.app.vocabularies.Users',
    required=True,
)
form.widget(
    'accountable',
    AjaxSelectFieldWidget,
    pattern_options={'allowNewItems': False},
)
```

Tested in Plone 5.0b2.

**Comment by vangheem** [comment]
Confirmed. Can you review and merge this PR that fixes this issue please? https://github.com/plone/mockup/pull/517

**Comment by pbauer** [comment]
fixed in #517
